dali/operators/reader/loader/webdataset_loader.cc

```diff
@@ -135,8 +184,9 @@ WebdatasetLoader::WebdatasetLoader(const OpSpec& spec)
       index_paths_(spec.GetRepeatedArgument<std::string>("index_paths")),
       missing_component_behavior_(detail::wds::ParseMissingExtBehavior(
           spec.GetArgument<std::string>("missing_component_behavior"))) {
-  DALI_ENFORCE(paths_.size() == index_paths_.size(),
-               "Number of webdataset archives does not match the number of index files");
+  DALI_ENFORCE(paths_.size() == index_paths_.size() || index_paths_.size() == 0,
+               "The number of index files, if any, must match the number of archives"
+               + "in the dataset.");
```

Comment:
```suggestion
               "in the dataset.");
```
The literal will be concatenated automatically. Also, This `+` only works as a side effect of how DALI_ENFORCE is implemented and may cease to work.